Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc

```diff
@@ -34,12 +34,11 @@ class MuonNumberingInitialization : public edm::ESProducer {
   ReturnType produce(const MuonNumberingRecord&);
 
 private:
-  edm::ESGetToken<DDCompactView, IdealGeometryRecord> geomToken_;
+  const edm::ESGetToken<DDCompactView, IdealGeometryRecord> geomToken_;
 };
 
-MuonNumberingInitialization::MuonNumberingInitialization(const edm::ParameterSet&) {
-  setWhatProduced(this).setConsumes(geomToken_);
-}
+MuonNumberingInitialization::MuonNumberingInitialization(const edm::ParameterSet&)
+    : geomToken_{setWhatProduced(this).consumes()} {}
```

Comment:
I believe
```suggestion
    : geomToken_{setWhatProduced(this)} {}
```
also works. Not sure I like it or not.

Reply:
Actually it doesn't work. I tested and it leads to compilation errors
```
In file included from .../src/FWCore/Framework/interface/es_impl/MayConsumeChooser.h:28,
                 from .../src/FWCore/Framework/interface/ESConsumesCollector.h:33,
                 from .../src/FWCore/Framework/interface/ESProducer.h:77,
                 from .../Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc:22:
.../FWCore/Utilities/interface/ESGetToken.h: In instantiation of 'constexpr edm::ESGetToken<ESProduct, ESRecord>::ESGetToken(ADAPTER&&) [with ADAPTER = edm::ESConsumesCollectorT<Mu
onNumberingRecord>; ESProduct = DDCompactView; ESRecord = IdealGeometryRecord]':
.../Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc:41:39:   required from here
.../FWCore/Utilities/interface/ESGetToken.h:52:117: error: no matching function for call to 'edm::ESConsumesCollectorT<MuonNumberingRecord>::consumes<DDCompactView, IdealGeometryRe
cord>()'
     constexpr explicit ESGetToken(ADAPTER&& iAdapter) : ESGetToken(iAdapter.template consumes<ESProduct, ESRecord>()) {}

In file included from .../FWCore/Framework/interface/es_impl/MayConsumeChooser.h:28,
                 from .../FWCore/Framework/interface/ESConsumesCollector.h:33,
                 from .../FWCore/Framework/interface/ESProducer.h:77,
                 from .../Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc:22:
.../FWCore/Utilities/interface/ESGetToken.h: In instantiation of 'constexpr edm::ESGetToken<ESProduct, ESRecord>::ESGetToken(ADAPTER&&) [with ADAPTER = edm::ESConsumesCollectorT<Mu
onNumberingRecord>; ESProduct = DDCompactView; ESRecord = IdealGeometryRecord]':
.../Geometry/MuonNumbering/plugins/MuonNumberingInitialization.cc:41:39:   required from here
.../FWCore/Utilities/interface/ESGetToken.h:52:117: error: no matching function for call to 'edm::ESConsumesCollectorT<MuonNumberingRecord>::consumes<DDCompactView, IdealGeometryRe
cord>()'
     constexpr explicit ESGetToken(ADAPTER&& iAdapter) : ESGetToken(iAdapter.template consumes<ESProduct, ESRecord>()) {}
```
I think this happens because `setWhatConsumes()` returns `ESConsumesCollector<T>`, whose `consumes()` takes only one template argument, while the `ESGetToken` constructor calls `consumes()` with two template arguments.

Anyway, I think the explicit call to `consumes()` makes the call slightly easier to understand.